simulator/datacenter.go

```diff
@@ -20,15 +20,40 @@ import (
 	"strings"
 
 	"github.com/vmware/govmomi/simulator/esx"
+	"github.com/vmware/govmomi/vim25/methods"
 	"github.com/vmware/govmomi/vim25/mo"
+	"github.com/vmware/govmomi/vim25/soap"
 	"github.com/vmware/govmomi/vim25/types"
 )
 
+type Datacenter struct {
+	mo.Datacenter
+
+	isESX bool
+}
+
+// NewDatacenter creates a Datacenter and its child folders.
+func NewDatacenter(f *Folder) *Datacenter {
+	dc := &Datacenter{
+		isESX: f.Self == esx.RootFolder.Self,
+	}
+
+	if dc.isESX {
+		dc.Datacenter = esx.Datacenter
```

Comment:
should there be a return here? I'm not sure if the `putChild` and `createFolders` are idempotent for ESX.
Doesn't look like `createFolders` is - looks like it'll overwrite an existing folder the in map.

Reply:
We need the same logic for both.  Using esx.Datacenter here just ends up using the ESX moref values such as `ha-datacenter`, instead of the VPX values.

Reply:
It looks like createFolders will create new folders on ESX even if the dc folders already exist - will that not remove any inventory held by those folders?

Reply:
The `NewDatacenter` method is only called once internally when vcsim is in ESX mode.  And the `Folder.CreateDatacenter` method is disabled in ESX mode.